### PR TITLE
Move and resize floating windows

### PR DESCRIPTION
**Pull request by Pamesta**

### CODE REVIEW

File: GlazeWM.Domain/Windows/CommandHandlers/ResizeWindowHandler.cs [hunk 1]
```diff
@@ -119,5 +125,37 @@ public CommandResponse Handle(ResizeWindowCommand command)
 
       return 1.0 / resizableLength;
     }
+    private void ResizeFloatingWindow(Window windowToResize, ResizeDimension dimensionToResize, string resizeAmount)
+    {
+      int MIN_WIDTH = (int)(windowToResize.Parent.Width * 0.30);
```
Comment (lars-berger) [inline GlazeWM.Domain/Windows/CommandHandlers/ResizeWindowHandler.cs:130]: perhaps a hardcoded pixel amount for min width/height instead (eg. 100 pixels)?

File: GlazeWM.Domain/Windows/CommandHandlers/ResizeWindowHandler.cs [hunk 2]
```diff
@@ -29,6 +30,11 @@ public CommandResponse Handle(ResizeWindowCommand command)
       var resizeAmount = command.ResizeAmount;
       var windowToResize = command.WindowToResize;
 
+      if (windowToResize is FloatingWindow)
+      {
+        ResizeFloatingWindow(windowToResize, dimensionToResize, resizeAmount);
+        return CommandResponse.Ok;
+      }
```
Comment (lars-berger) [inline GlazeWM.Domain/Windows/CommandHandlers/ResizeWindowHandler.cs:37]: think it'd be good to have a `ResizeTilingWindow` method

File: GlazeWM.Domain/UserConfigs/GeneralConfig.cs [hunk 3]
```diff
@@ -11,5 +11,9 @@ public class GeneralConfig
     /// TODO: Not officially released because implementation is buggy. Use at own risk.
     /// </summary>
     public bool CursorFollowsFocus { get; set; }
+    /// <summary>
+    /// Amount by which to move floating windows
+    /// </summary>
+    public int FloatingWindowMoveAmount { get; set; } = 20;
```
Comment (lars-berger) [inline GlazeWM.Domain/UserConfigs/GeneralConfig.cs:17]: would be good to change this to a string instead, so users could use either `20` or `20px` (then use `UnitsHelper.TrimUnits()` to get an int). also make it easier to allow % values if we add that in the future

File: GlazeWM.Domain/Windows/CommandHandlers/MoveWindowHandler.cs [hunk 4]
```diff
@@ -216,5 +232,37 @@ private void ChangeWorkspaceLayout(Window windowToMove, Direction direction)
...
+          break;
+
+        case Direction.Up:
+          y -= amount;
+          break;
+
+        case Direction.Down:
+          y += amount;
+          break;
+      }
```
Comment (lars-berger) [inline GlazeWM.Domain/Windows/CommandHandlers/MoveWindowHandler.cs:260]: can be simplified with
```csharp
      _ = direction switch
      {
        Direction.Left => x -= amount,
        Direction.Right => x += amount,
        Direction.Up => y -= amount,
        Direction.Down => y += amount,
      };
```

File: .vscode/launch.json [hunk 5]
```diff
@@ -8,7 +8,8 @@
       "name": ".NET Core Launch (console)",
       "type": "coreclr",
       "request": "launch",
-      "preLaunchTask": "build",
+      "enableStepFiltering": false,
```
[the remark this answers is not in the document]
Comment (Pamesta) [reply .vscode/launch.json:11]: sorry, modified it for myself and forgot to restore before pushing

**Comment by Pamesta** [comment]
Thank you, going to clean up and edit the code based on the comments

**Pull request merged**